fhirstore/fhirstore.py

```diff
@@ -240,14 +254,58 @@ def search(self, resource_type, params):
             sub_query = build_simple_query(params)
         elif len(params) > 1:
             inter_query = [
-                build_simple_query({sub_key: sub_value}) for sub_key, sub_value in params.items()
+                build_simple_query({sub_key: sub_value})
+                for sub_key, sub_value in params.items()
             ]
             sub_query = {"bool": {"must": inter_query}}
-        query = {"min_score": 0.01, "query": sub_query}
 
+        query = {
+            "min_score": 0.01,
+            "from": offset,
+            "size": result_size,
+            "query": sub_query,
+        }
+
+        if elements:
+            query["_source"] = elements
         # .lower() is used to fix the fact that monstache changes resourceTypes to
         # all lower case
         hits = self.es.search(body=query, index=f"fhirstore.{resource_type.lower()}")
-        response = {"resource_type": "Bundle", "items": hits["hits"]["hits"]}
+        bundle = {
+            "resource_type": "Bundle",
+            "items": [h["_source"] for h in hits["hits"]["hits"]],
+            "total": hits["hits"]["total"]["value"],
+        }
+        if elements:
+            bundle["tag"] = {"code": "SUBSETTED"}
+
+        return bundle
+
+    def count(self, resource_type, params):
+        """Counts how many results match to this query
+        """
+        self.validate_resource_type(resource_type)
+        validate_parameters(params)
+
+        sub_query = defaultdict(lambda: defaultdict(dict))
+        if len(params) == 0:
+            sub_query = {"match_all": {}}
+        elif len(params) == 1:
+            sub_query = build_simple_query(params)
+        elif len(params) > 1:
+            inter_query = [
+                build_simple_query({sub_key: sub_value})
+                for sub_key, sub_value in params.items()
+            ]
+            sub_query = {"bool": {"must": inter_query}}
+
+        query = {"query": sub_query}
 
-        return response
+        # .lower() is used to fix the fact that monstache changes resourceTypes to
+        # all lower case
+        hits = self.es.count(body=query, index=f"fhirstore.{resource_type.lower()}")
+        return {
+            "resource_type": "Bundle",
+            "tag": {"code": "SUBSETTED"},
+            "total": hits["count"],
+        }
```

Comment:
Maybe I'm anticipating a bit too much but maybe you could have a function or something where you will handle all possible summaries (not asking to implement everything right now but just to make some room for future work) at the end of which you would add the SUBSETTED tag

Reply:
Ok, I'll write an issue for this